app/components/layers/useCapitalProjectBudgetedGeoJsonLayer.client.tsx

```diff
@@ -0,0 +1,34 @@
+import { GeoJsonLayer, GeoJsonLayerProps } from "@deck.gl/layers";
+import { CapitalProjectBudgetedGeoJson } from "~/gen";
+import { useParams } from "@remix-run/react";
+import { FlyToGeoJsonExtension } from "../../extensions";
+
+export function useCapitalProjectBudgetedGeoJsonLayer() {
+  const { managingCode, capitalProjectId } = useParams();
+
+  return new GeoJsonLayer<CapitalProjectBudgetedGeoJson["properties"]>({
+    id: "capitalProjectBudgetedGeoJson",
+    data:
+      managingCode === undefined || capitalProjectId === undefined
+        ? []
+        : `${import.meta.env.VITE_ZONING_API_URL}/api/capital-projects/${managingCode}/${capitalProjectId}/geojson`,
+    pickable: true,
+    getFillColor: ({ properties }) => {
+      const { managingCode: selectedProjectManagingCode, id } = properties;
+      switch (`${selectedProjectManagingCode}${id}`) {
```

Comment:
We were more strict than the geojson spec and made the Id a required/guaranteed value in [our geojson endpoints](https://staging-zoning-api.nycplanningdigital.com/#tag/GeoJSON/operation/findCapitalProjectGeoJsonByManagingCodeCapitalProjectId). It's for just this scenario when we want the id already concatenated.

```suggestion
    
    getFillColor: ({ id }) => {
      switch (id) {
```